### PR TITLE
Improve `redis/asyncio/client.pyi`

### PR DESCRIPTION
**Pull request by sobolevn**
There are other things that can be improved. But, I went with types I am 100% sure.
Source: https://github.com/redis/redis-py/blob/master/redis/asyncio/client.py

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Good to see you around these parts again, @sobolevn!

File: stubs/redis/redis/asyncio/client.pyi [hunk 1]
```diff
@@ -22,12 +22,12 @@ ResponseCallbackT: TypeAlias = ResponseCallbackProtocol | AsyncResponseCallbackP
...
     response_callbacks: MutableMapping[str | bytes, ResponseCallbackT]
-    @classmethod
-    def from_url(cls, url: str, **kwargs) -> Redis[Any]: ...
-    auto_close_connection_pool: Any
+    auto_close_connection_pool: bool
     connection_pool: Any
     single_connection_client: Any
     connection: Any
+    @classmethod
+    def from_url(cls, url: str, **kwargs) -> Redis[Any]: ...
```
Comment (AlexWaygood) [inline stubs/redis/redis/asyncio/client.pyi:30]: Why move this method down a few lines? It doesn't look like the signature's changed.

File: stubs/redis/redis/asyncio/client.pyi [hunk 2]
```diff
@@ -116,15 +116,15 @@ class Monitor:
     async def listen(self) -> AsyncIterator[MonitorCommandInfo]: ...
 
 class PubSub:
-    PUBLISH_MESSAGE_TYPES: Any
-    UNSUBSCRIBE_MESSAGE_TYPES: Any
+    PUBLISH_MESSAGE_TYPES: tuple[str, ...]
+    UNSUBSCRIBE_MESSAGE_TYPES: tuple[str, ...]
     HEALTH_CHECK_MESSAGE: str
```
Comment (AlexWaygood) [inline stubs/redis/redis/asyncio/client.pyi:121]: These three should probably be `ClassVar`s

File: stubs/redis/redis/asyncio/client.pyi (hunk 1 above)
Comment (sobolevn) [reply stubs/redis/redis/asyncio/client.pyi:30]: So attributes and methods are not mixed. It was hard to read.

Comment (AlexWaygood) [reply stubs/redis/redis/asyncio/client.pyi:30]: Okay, fair enough :)

File: stubs/redis/redis/asyncio/client.pyi [hunk 3]
```diff
@@ -201,5 +201,5 @@ class Pipeline(Redis[_StrType], Generic[_StrType]):
     async def load_scripts(self) -> None: ...
     async def execute(self, raise_on_error: bool = ...): ...
     async def discard(self) -> None: ...
-    async def watch(self, *names: KeyT): ...
-    async def unwatch(self): ...
+    async def watch(self, *names: KeyT) -> bool: ...
```
[the remark this answers is not in the document]
Comment (sobolevn) [reply stubs/redis/redis/asyncio/client.pyi:204]: In this partical case command `WATCH` always returns `OK` from `redis` which is translated to `True`:

```python
>>> from redis.asyncio import Redis
>>> redis = Redis.from_url('redis://localhost:6379')
>>> import asyncio
>>> async def main():
...   p = await redis.pipeline()
...   print('got', await p.watch('a'))
... 
>>> asyncio.run(main())
got True
```

[the remark this answers is not in the document]
Comment (AlexWaygood) [reply stubs/redis/redis/asyncio/client.pyi:204]: Got it, thanks.

**Review by AlexWaygood — APPROVED** [review]

**Review by orsinium — APPROVED** [review]

**Pull request merged**